src/main/java/se/liu/ida/hefquin/engine/datastructures/impl/SolutionMappingsHashTableBasedOnOneVar.java

```diff
@@ -0,0 +1,89 @@
+package se.liu.ida.hefquin.engine.datastructures.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
+
+import java.util.*;
+
+public class SolutionMappingsHashTableBasedOnOneVar extends SolutionMappingsHashTable{
+    protected final Map<Node, List<SolutionMapping>> map = new HashMap<>();
+    protected final Var joinVar;
+
+    public SolutionMappingsHashTableBasedOnOneVar(final Set<Var> joinVariables) {
+        if (joinVariables.size() != 1){
+            throw new IllegalArgumentException();
+        }else {
+            this.joinVar = joinVariables.iterator().next();
+        }
+    }
+
+    public SolutionMappingsHashTableBasedOnOneVar(final Var... vars) {
+        if (vars.length != 0) {
+            throw new IllegalArgumentException();
+        }
+        this.joinVar = vars[0];
+    }
+
+    @Override
+    public boolean add( final SolutionMapping e ) {
+        final Node valKeys = getVarKey(e);
```

Comment:
There needs to be check whether getVarKey returns null. If it does, throw an IllegalArgumentException.